Mathlib/Lean/CoreM.lean

```diff
@@ -33,3 +34,23 @@ def reportOutOfHeartbeats (tac : Name) (stx : Syntax) (threshold : Nat := 90) :
   if (← heartbeatsPercent) ≥ threshold then
     logInfoAt stx (s!"`{tac}` stopped because it was running out of time.\n" ++
       "You may get better results using `set_option maxHeartbeats 0`.")
+
+/--
+Term elaborator that retrieves the current `SearchPath`.
+-/
+elab "compileTimeSearchPath" : term =>
+  return toExpr (← searchPathRef.get)
+
+/--
+Run a `CoreM α` in a fresh `Environment` with specified `modules : List Name` imported.
+-/
+def CoreM.withImportModules (modules : List Name) (run : CoreM α)
+    (searchPath : Option SearchPath := none) (options : Options := {})
+    (trustLevel : UInt32 := 1024) (fileName := "") :
+    IO α := unsafe do
+  searchPathRef.set (searchPath.getD compileTimeSearchPath)
```

Comment:
The output of `compileTimeSearchPath` is build host-dependent and should not be baked into compiled oleans (it includes amongst other things the full home directory path).  This would break running the scripts locally (or maybe even on a different CI machine than the one which compiled the `CoreM.lean` file).

Reply:
In case it's not clear, `compileTimeSearchPath` is an elaborator and not a definition.

Reply:
Thanks. Could you take another look?